aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -328,6 +328,41 @@ void add_dense_sparse_worker_cpu(Tensor& r, Scalar value, const SparseTensor& sp
   });
 }
 
+template <typename scalar_t>
+void add_dense_sparse_worker_cpu(Tensor& r, Scalar value, const SparseTensor& sparse, const Tensor& indices, const Tensor& values) {
+
+  int64_t values_dense_size = values.stride(0);
+  AT_ASSERT(values_dense_size > 0);
+  AT_ASSERT(values.is_contiguous());
+
+  auto indices_accessor = indices.accessor<int64_t, 2>();
+
+  scalar_t* r_ptr = r.data<scalar_t>();
+  AT_ASSERT(r_ptr != nullptr);
+  auto r_offset = r.storage_offset();
+
+  scalar_t* v_ptr = values.data<scalar_t>();
+  AT_ASSERT(v_ptr != nullptr);
+  auto v_offset = values.storage_offset();
+
+  scalar_t cast_value = value.to<scalar_t>();
+  at::parallel_for(0, sparse._nnz(), 0, [&](int64_t start, int64_t end) {
+    for (auto k = start; k < end; k++) {
+      auto r_index = r_offset;
+      for (int64_t d = 0; d < sparse.sparse_dim(); d++) {
+        r_index += r.stride(d) * indices_accessor[d][k];
+      }
+      auto v_index = v_offset + k * values_dense_size;
+      //THBlas_axpy<scalar_t>(values_dense_size, cast_value, v_ptr + v_index, 1, r_ptr + r_index, 1);
```

Comment:
Remove comments

Reply:
@VitalyFedyunin I just wonder whether function THBlas_axpy<scalar_t>(...) can be called inside the parallel region which can help to do vectorization for the for(...) loop. Or whether "#pragma omp simd" can be used here ?